dashboard/app/lib/javascripts/dashboard/main.js

```diff
@@ -110,12 +201,17 @@ window.Dashboard = {
 				this.__handleGithubAuthChange(event.authenticated);
 			break;
 
+			case "CONTROLLER_UNREACHABLE_FROM_HTTPS":
+				// Controller isn't accessible via https. Redirect to http and try again.
+				window.location.href = window.location.href.replace("https", "http");
+			break;
+
+			case "HTTPS_CERT_MISSING":
+				Marbles.history.navigate("installcert");
```

Comment:
`Marbles.history.navigate("installcert", {force: true});` so the handler is called even if the `window.location.href` already matches.